surveyor.py

```diff
@@ -5,477 +5,457 @@
+                # run search based on IOC file
 
+                    log_echo(f"Processing IOC list: {ioc_source}", self._log)
+                    ioc_list = [x.strip() for x in ioc_list]
+
+                    product.nested_process_search(Tag(f"IOC - {ioc_source}", data=ioc_source), {ioc_type: ioc_list}, base_query)
 
-table_template: Tuple[int, int, int, int, int, int] = (30, 30, 30, 30, 30, 30)
-table_template_str = f'{{:<{table_template[0]}}} ' \
-                     f'{{:<{table_template[1]}}} ' \
-                     f'{{:<{table_template[2]}}} ' \
-                     f'{{:<{table_template[3]}}}'
+                    for tag, results in product.get_results().items():
+                        self._write_results(results, ioc_source, 'ioc', tag)
+
+            # Sigma
+            sigma_rules = list()
+            if sigma_rule or sigma_rules or sigma_rules_str:
+                 
+                if sigma_rule:
+                    sigma_rules.append(sigma_rule)
+
+                #If list of one or more sigma rules is provided, rule(s) will be translated.
+                if sigma_rules:
+                    translated_rules = sigma_translation(product=self._edr, sigma_rules=sigma_rules, file=True)
+                    length = len(sigma_rules)
+
+            #If str of yaml containing one or more sigma rules is provided, rule(s) will be translated.
+                elif sigma_rules_str:
+                    translated_rules = sigma_translation(product=self._edr, sigma_rules=[sigma_rules_str], file=False)
+                    length = len(translated_rules['queries'])
+
+                if len(translated_rules['queries']) != length:
+                    self._log.warning(f"Only {len(translated_rules['queries'])} out of {length} were able to be translated.")
+
+                for rule in tqdm(translated_rules['queries'], desc="Processing sigma rules", disable=no_progress):
+                    program = f"{rule['title']} - {rule['id']}"
+                    source = 'Sigma Rule'
+
+                    product.nested_process_search(Tag(program, data=source), {'query': [rule['query']]}, base_query)
+
+                    if product.has_results():
+                        # write results as they become available
+                        for tag, nested_results in product.get_results(final_call=False).items():
+                            self._write_results(nested_results, program, str(tag.data), tag)
+
+                        # ensure results are only written once
+                        product.clear_results()
+
+                # write any remaining results
+                for tag, nested_results in product.get_results().items():
+                    self._write_results(nested_results, tag.tag, str(tag.data), tag)
+
+            # Definition Files
+
+            # run search against definition files and write to csv
+            if definitions:
+                if isinstance(definitions,list):
```

Comment:
if we're concerned that the variable `definitions` won't be a list, i think we should close this with an `else` statement and raise an error